applications/autonomous_worlds_on_substrate.md

```diff
@@ -0,0 +1,346 @@
+| 5. | Pallet: The "beacon" Pallet (1 week) | We develop and deploy a pallet that accepts interpolated signaures and proofs, following the same logic as the beacon contract. We include this within our runtime. |
+| 6. | Demo Contract: Rock Paper Scissors **Part 2** (1 week) | We develop a rock-paper-scissors demo contract that consumes randomness from the SCLC and uses it in a meaningful way (to play p2p non-interactive RPS). |
+
+### Milestone 2: Practical Witness Encryption Implementation
+
+- **Estimated Duration:** 4 weeks
+- **FTE:** 3
+- **Costs:** 3300 DOT (6.875 DOT/hour)
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Specify the IP owning entity; ensure proper copyright compliance for all reused materials, including appropriate licenses and attributions, or choose an open-source license (Apache 2.0, GPLv3, MIT, Unlicense). |
+| **0b.** | Documentation | Provide comprehensive **inline documentation** of the code and a detailed **tutorial**. The tutorial should guide users on how to set up, play the game, and assess the milestone's deliverables, ensuring functionality and compliance with the milestone objectives. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Platform | Provide a detailed description of the game's platform compatibility and the infrastructure setup required to host and run the game, including supported operating systems, hardware requirements, and necessary backend services. |
+| 0e. | Article | We will publish an **article**/workshop that explains [...] (what was done/achieved as part of the games bounty). (Content, language, and medium should reflect your target audience described above.) |
+| 1. | Library: Etf.js - SnarkJS support | We develop a simple circuit library and add pre-compiles to IPFS. Then we modify the etf.js library to use the [snarkJS](https://github.com/iden3/snarkjs) library to prepare witness files with new inputs on pre-compiled circuits.   |
+| 2. | Substrate Runtime: pallet-witness-encryption (part 1) | We develop a new pallet for enabling practical on-chain witness encryption. The pallet functions as described above, where users encode public parameters for circuits and resharings of the associated secret in a pallet, which is then associated with a new on-chain asset class. We adapt this from the work done by [bright](https://brightinventions.pl/blog/zk-snarks-in-substrate-part-1/), using [this zkSnark pallet](https://github.com/bright/zk-snarks-with-substrate/tree/main/pallets/zk-snarks) to get started. Note: This intermediate step does not modify validator logic yet. We leverage the acss scheme already implemented in the Ideal network [here](https://github.com/ideal-lab5/etf-sdk/blob/w3fbls-migration/etf-crypto-primitives/src/dpss/acss.rs). |

```

Comment:
I would leave the details out of this section. Maybe move it to the main body of the Milestone 2

Reply:
Good call, I'll clean it up and make it a single task to develop that pallet.